protected/views/search/_new_result.php

```diff
@@ -1,73 +1,102 @@
-    <?php
+    <?php foreach ($datasets['data'] as $dt) { ?>
+        <?php
         $filterOnDatasetId = function ($var) use ($dt) {
             return $var['dataset_id'] === $dt['id'];
         };
         $dsamples = array_filter($samples['data'], $filterOnDatasetId);
-        $dfiles = array_filter($files['data'], $filterOnDatasetId) ;
-    ?>
-    <div class="search-result-container">
-        <!--Dataset section-->
-        <?php $is_display = in_array('dataset', $display) || (in_array('file', $display) && $dfiles) || (in_array('sample', $display) && $dsamples); ?>
-        <?php if($is_display) { ?>
-        <div class="row1">
-            <div class="span1" style="margin-left: 40px;margin-top: 10px;height: 30px;width: 40px"><div class="text-icon text-icon-sm text-icon-blue" style="margin-right: 0px;">G
-            </div></div>
-            <div class="span8 main-content" style="float:right">
-                <ul class="nav nav-tabs nav-stacked result-cell">
-                  <li><a data-content="<?php echo CHtml::encode($dt['description']) ?>" class="result-main-link left content-popup" href="<?php echo $dt['shorturl'] ?>"><?php echo $dt['title'] ?></a></li>
-                  <li>
-                    <strong>
-                        <?php echo $dt['authornames'] ?>
-                    </strong>
-                  </li>
-                  <li class="searchID"><?= Yii::t('app', 'DOI') ?>:<?php echo "10.5524/" . $dt['identifier'] ?></li>
-                </ul>
-            </div>
-        </div>
-        <?php } ?>
+        $dfiles = array_filter($files['data'], $filterOnDatasetId);
+        $is_display = in_array('dataset', $display) || (in_array('file', $display) && $dfiles) || (in_array('sample', $display) && $dsamples);
+        ?>
+        <?php if ($is_display) { ?>
+            <div class="search-result-card">
+                <!--Dataset section-->
+                <div class="search-result-row">
+                    <div class="search-result-icon-container">
+                        <div class="text-icon text-icon-sm text-icon-blue search-result-icon" aria-hidden="true">G</div>
+                        <span class="sr-only">Dataset</span>
+                    </div>
+                    <div class="search-result-content">
+                        <h3 class="search-result-title">
+                            <a data-content="<?php echo CHtml::encode($dt['description']) ?>" class="search-result-link content-popup" href="<?php echo $dt['shorturl'] ?>"><?php echo $dt['title'] ?></a>
+                        </h3>
+                        <?php if (!empty($dt['authornames'])) : ?>
+                            <!-- HACK to render a list of links for authors, since the model directly returns a sequence of links as a string -->
+                            <div class="search-result-subcontent">
+                                <?php
+                                    $authorLinks = explode(';', $dt['authornames']);
```

Comment:
`$dt['authornames']` value is something like `<a class=result-sub-links href=/search/new?keyword=Zhang,G&author_id=3325>Zhang G</a>; <a class=result-sub-links href=/search/new?keyword=Lambert,DM&author_id=3371>Lambert DM</a>; <a class=result-sub-links href=/search/new?keyword=Wang,J&author_id=3257>Wang J</a>`

I parsed it with php to turn it into a <ul>

Perhaps it would be more resilient to get the list of author objects with the appropriate properties (href, label) and let this file determine how it is rendered

If this sounds like a good idea, I could try to implement it

Reply:
Hi @luistoptal


I agree with you in principle, but I don't think it's worth the effort.
The linked list of authors is actually created in the PostgreSQL search index builder [1].
Our unique goal when we started making use of PostgreSQL Full Text Search, was to replace the dead-slow predecessor with a very fast search that can be retrofitted into the existing Search UX.
That meant, among other limitations,  making sure all dynamic data displayed in UI is indexed and returned from the search query.
The end result is we were able to implement a very fast search without changing much of the UI but at the price of some rigidity in data presentation, which I was grudgingly willing to accept at the time.

We have plan to improve the search backend because it was designed to work with PostgreSQL 11, and we are now on PostgreSQL 14 which has new search features that can simplify our implementation.
I'd rather wait until we get to do that work to make the change you are suggesting.
 
[1] https://github.com/gigascience/gigadb-website/blob/develop/sql/dataset_finder.sql